SocketIOClientSwift/SocketEngine.swift

```diff
@@ -629,26 +629,22 @@ public final class SocketEngine: NSObject, WebSocketDelegate {
 
             ws?.writeString("\(type.rawValue)\(str)")
 
-            if let datas = datas {
-                for data in datas {
-                    let (data, _) = createBinaryDataForSend(data)
-                    if data != nil {
-                        ws?.writeData(data!)
-                    }
+            for data in datas ?? [] {
+                let (data, _) = createBinaryDataForSend(data)
+                if data != nil {
+                    ws?.writeData(data!)
                 }
             }
     }
 
     // Starts the ping timer
     private func startPingTimer() {
-        guard pingInterval != nil else {
-            return
-        }
+        if let pingInterval = pingInterval {
```

Comment:
I prefer to avoid early returns that can be problematic if you don't notice them in the future. But I understand that's an stylistic choice (unlike some other things in this PR), so I'll chose whatever you guys prefer.